### PR TITLE
Integrate bubble correction

### LINKED ISSUE #1211: Integrate bubble correction
Let it bubble! 

The aim of this issue is to track progress of the integration of the bubble correction feature from ClimaCore into ClimaAtmos. 

We want to: 

- [x] Modify the driver
- [ ] Make it default (?) To be discussed after seeing test results.
- [x] Modify the integration tests pipeline to test the new feature

### PR DESCRIPTION
**Pull request by valeriabarra**
## Purpose 
This PR integrates the bubble correction feature from ClimaCore into ClimaAtmos.

Closes #1211 

## Content
- [x] Added a dedicated CLI option, called `--bubble` (for now) false by default
- [x] Modified driver to include new functionality in the suitable configurations (i.e., not in column configuration), and added warning when used with non suitable configurations.
- [x] Modified regular and longrun pipelines (most tests including bubble will be those with Float64, as with Float32 this functionality does not provide much benefit, if at all).

Review checklist

I have:
- followed the codebase contribution guide: https://clima.github.io/ClimateMachine.jl/latest/Contributing/
- followed the style guide: https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/
- followed the documentation policy: https://github.com/CliMA/policies/wiki/Documentation-Policy
- checked that this PR does not duplicate an open PR.

In the Content, I have included 
- relevant unit tests, and integration tests, 
- appropriate docstrings on all functions, structs, and modules, and included relevant documentation.

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Review by charleskawczynski — COMMENTED** [review]
The code changes look fine to me. I haven't looked at how the solutions change (none of the jobs have mse tables). Also, why does this significantly impact the `baroclinic wave (ρe_tot) equilmoist check conservation float64` test?

**Comment by valeriabarra** [comment]
> The code changes look fine to me. I haven't looked at how the solutions change (none of the jobs have mse tables). Also, why does this significantly impact the `baroclinic wave (ρe_tot) equilmoist check conservation float64` test?

Yeah, I am surprised that this is actually making conservation worse. I am investigating again in ClimaCore. Trying to reproduce on simpler examples.

**Comment by valeriabarra** [comment]
Currently investigating this in ClimaCore (see relevant [issue](https://github.com/CliMA/ClimaCore.jl/issues/1063)). I have temporarily put this in Draft state until this is cleared in ClimaCore.

File: examples/hybrid/driver.jl [hunk 1]
```diff
@@ -394,13 +394,18 @@ if parsed_args["regression_test"]
...
+    @test abs(sum(sol.u[1].c.ρ) - sum(sol.u[end].c.ρ)) / sum(sol.u[1].c.ρ) ≈
+          FT(0) atol = 4 * eps(FT)
+    @test abs(
+        sum(sol.u[1].c.ρe_tot) +
+        (p.net_energy_flux_sfc[][] - p.net_energy_flux_toa[][]) -
+        sum(sol.u[end].c.ρe_tot),
+    ) / (
+        sum(sol.u[1].c.ρe_tot) +
+        (p.net_energy_flux_sfc[][] - p.net_energy_flux_toa[][])
+    ) ≈ FT(0) atol = 2 eps(FT)
```
Comment (valeriabarra) [inline examples/hybrid/driver.jl:406]: @simonbyrne here is where I found that keeping the test with `atol` passes (and with much smaller tolerances than the original version), but not with `rtol`

Comment (simonbyrne) [reply examples/hybrid/driver.jl:406]: So `isapprox(x, y; atol)` will check absolute error
```
norm(x - y) < atol
```
and `isapprox(x, y; rtol)` does relative error
```
norm(x - y) < rtol * max(norm(x), norm(y))
```

What you're doing is trying manually computing the relative error and comparing it to zero: i.e. you're doing
```
isappox((x - y)/x, 0)
```
where `x = sum(sol.u[1].c.ρe_tot) + (p.net_energy_flux_sfc[][] - p.net_energy_flux_toa[][])` and `y = sum(sol.u[end].c.ρe_tot)`, and so the error has already been "relativized"

Probably what you want is
```suggestion
    @test sum(sol.u[1].c.ρ) ≈ sum(sol.u[end].c.ρ)) rtol = 4 * eps(FT)
    @test sum(sol.u[1].c.ρe_tot) + (p.net_energy_flux_sfc[][] - p.net_energy_flux_toa[][]) ≈
        sum(sol.u[end].c.ρe_tot) rtol = 2 eps(FT)
```

**Comment by valeriabarra** [comment]
This passes the conservation check after the suggested fix in ClimaCore (see successful [build]( https://buildkite.com/clima/climaatmos-ci/builds/6383)). As soon as we tag the new release of ClimaCore, I will update this branch Manifest

**Comment by valeriabarra** [comment]
This PR is ready to merge. CI passes

**Pull request merged**